lib/datadog/core/configuration/options.rb

```diff
@@ -84,6 +84,12 @@ def option_defined?(name)
             self.class.options.key?(name)
           end
 
+          def default_option?(option)
+            return options[option].default_precedence? if options[option]
+
+            !self.class.options[option].default.nil?
```

Comment:
I thought about it more and leaving it as it made sense. 

In the case of an option with a default value that has yet to be accessed. It makes sense to return `true` since the next time the option is accessed, the returned value is the default value.  If we have set a value by any change, the result from calling `default_option?` would be false.

In the case of an option with a default value that has yet to be accessed. It makes sense to return `false` since there is no default set.